test/test_environment.rb

```diff
@@ -746,6 +746,19 @@ def foo; end
     @env.unregister_preprocessor('application/javascript', processor)
     assert_equal "// =require \"notfound\"\n;\n", @env["missing_require.js"].to_s
   end
+
+  test "disabling processors by class name" do
+    class SpaceAndWhitespaceCompressor < WhitespaceCompressor; end
```

Comment:
Can you define this class inline? 

Like `SpaceAndWhitespaceCompressor = Class.new(WhitespaceCompressor)`